cmd/cronjob-checker/main.go

```diff
@@ -22,16 +22,25 @@ var kubeConfigFile = filepath.Join(os.Getenv("HOME"), ".kube", "config")
 // namespace is a variable to allow code to target all namespaces or a single namespace
 var namespace = os.Getenv("NAMESPACE")
 
+// Check deadline from injected env variable KH_CHECK_RUN_DEADLINE
+var khCheckRunDeadlineEnv = os.Getenv("KH_CHECK_RUN_DEADLINE")
+
 func init() {
 	// set debug mode for nodeCheck pkg
 	nodeCheck.EnableDebugOutput()
 }
 
 func main() {
+	intkhCheckRundDeadline, err := strconv.ParseInt(khCheckRunDeadlineEnv, 10, 64)
+	if err != nil {
+		log.Fatalln("Error parsing KH_CHECK_RUN_DEADLINE:", err)
+	}
+	khCheckRunDeadline := time.Unix(intkhCheckRundDeadline, 0)
+
+	khCheckRunTime := khCheckRunDeadline.Sub(time.Now()) / 2
 
-	// create context
-	checkTimeLimit := time.Minute * 1
-	ctx, _ := context.WithTimeout(context.Background(), checkTimeLimit)
+	log.Infoln(khCheckRunDeadline)
```

Comment:
For production code, let's either drop this log line.  Also, leaving a comment on the check run deadline code above would be nice.

Reply:
removed and comment added